hermes/file_gen.go

```diff
@@ -1,51 +1,70 @@
+//
 // Unless required by applicable law or agreed to in writing, software
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
+//
 // Author: Alicja Kwiecinska (kwiecinskaa@google.com) github: alicjakwie
+//
+// File_gen provides file generation methods for Hermes.
 
-// File generation for Hermes
+// Package hermes is the general package for methods and function used within Hermes.
 package hermes
 
 import (
 	"crypto/sha1"
 	"fmt"
 )
 
+// HermesFile represents a file that Hermes will create on the storage system.
 type HermesFile struct {
 	name     string
 	contents string
 }
 
-// method of HermesFile generates HermesFile.name of the form Hermes_id_checksum where id is an integer & id <= 50
-func (file *HermesFile) generateFileName(file_id int, file_checksum string) {
-	file.name = fmt.Sprintf("Hermes_%02d_%v", file_id, file_checksum)
+// generateFileName generates name of the form Hermes_ID_checksum.
+// where ID is an integer & ID <= 49 & ID >= 0.
+// Arguments:
+//	- fileID: ID must be >= 0 and <= 49.
+//	- fileChecksum: checksum must be a SHA1 checksum in hex notation.
+func (f *HermesFile) generateFileName(fileID int, fileChecksum string) {
+	f.name = fmt.Sprintf("Hermes_%02d_%v", fileID, fileChecksum)
 }
 
-// method of HermesFile generates HermesFile.contents now a string without any significance in the future a pseudo random byte generator will be used
-func (file *HermesFile) generateFileContents(file_id int) {
-	file.contents = "jhfvjhdfjhfjjhjhdfvjvcvfjh"
+// generateFileContents generates the contents of a file.
+// TODO(alicjakwie): Use a pseudo-random number generator instead.
+// Arguments:
+//	- fileID: the ID of the file. Must be between 0 and 49.
+func (f *HermesFile) generateFileContents(fileID int) {
+	f.contents = "jhfvjhdfjhfjjhjhdfvjvcvfjh"
 }
 
-// method of HermesFile generates the checksum of the file contents
-func (file HermesFile) generateFileChecksum() string {
-	file_contents := []byte(file.contents)
-	hash := sha1.Sum(file_contents)
+// generateFileChecksum generates the checksum of the file contents.
+// Returns:
+//	- checksum: Returns the checksum of the file contents in hex notation.
+func (f *HermesFile) generateFileChecksum() string {
+	fileContents := []byte(f.contents)
+	hash := sha1.Sum(fileContents)
 	// return checksum in hex notation
 	return fmt.Sprintf("%x", hash)
 }
 
-// method of HermesFile generates the file takes id as a parameter
+// GenerateHermesFile generates a HermesFile object with a valid file name and contents.
+// Arguments:
+//	- id: id must be >= 0 and <= 49.
+// Returns:
+//	- HermesFile: returns a HermesFile object with a valid file name and contents.
 func GenerateHermesFile(id int) HermesFile {
 	file := HermesFile{}
 	file.generateFileContents(id)
```

Comment:
Not that this is wrong, but I think it's cleaner to have a stand-alone function that generates the contents, then this would be 

`file.contents = generateContents(id)`

...which reads better. The current version hides what's actually happening.

Reply:
I had this in a previous version but during a review was told that it would be better to make generateFileContents and generateFileName methods of the struct HermesFile. What is in your opinion more readable?

Reply:
The name "generateFileContents" suggests that its output is the contents.
However, the current version also sets the contents on the file, which is surprising.

Methods make sense when they access one or more fields of the receiver as an input or they use a mutex in the receiver.
generateFileContents() doesn't really use the file as an input.

Reply:
Okay thank you! will go back to the previous version then.

Reply:
@alicjakwie is making these changes in a different PR: https://github.com/googleinterns/step224-2020/pull/58 so I'll mark this resolved for now.